File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        reverse(digits.begin(),digits.end());
        int carry=0;
        vector<int>ans;
        for(int i=0;i<digits.size();i++){
            int sum=0;
            if(i==0){
              sum=digits[i]+1;

            }
            else{
                sum=digits[i]+carry;
            }
            ans.insert(ans.begin(),sum%10);
            carry=sum/10;
        }
        if(carry){
            ans.insert(ans.begin(),carry);
        }
        // reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: 0066-plus-one/0066-plus-one.rs
impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let mut ans : Vec<i32> = Vec::new();
        let mut carry = 0;
        
        for i in (0..digits.len()).rev() {
            let mut val;
            if i == digits.len() -1 {
                val = digits[i] + carry + 1;
            }
            else {
                val = digits[i] + carry ;
            }
            ans.insert(0,val%10);
            carry=val/10;
        }
        if carry != 0 {
            ans.insert(0,carry);
        }
        return ans;
    }
}